<a href="https://colab.research.google.com/github/Nauviii/first_project_DataAnalyst/blob/main/Notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Proyek Analisis Data: Bike Sharing Dataset
- **Nama:** Fikri Kurnia
- **Email:** ds.fikri.kurnia@gmail.com
- **ID Dicoding:** https://www.dicoding.com/users/fikrikurnia/academies

## Menentukan Pertanyaan Bisnis

- Pertanyaan 1 : Bagaimana pengaruh musim (musim panas, dingin, gugur, semi) terhadap jumlah sepeda yang dipinjam?
- Pertanyaan 2 : Apakah jumlah sepeda yang dipinjam lebih tinggi pada hari kerja dibandingkan akhir pekan?

## Import Semua Packages/Library yang Digunakan

In [66]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Data Wrangling

In [67]:
!git clone https://github.com/Nauviii/first_project_DataAnalyst
file_path_day = "first_project_DataAnalyst/dataset/day.csv"
file_path_hour = "first_project_DataAnalyst/dataset/hour.csv"


fatal: destination path 'first_project_DataAnalyst' already exists and is not an empty directory.


### Gathering Data

In [68]:
bike_day_df = pd.read_csv(file_path_day)
display(bike_day_df.head(10))

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1        0        6           0   
1        2  2011-01-02       1   0     1        0        0           0   
2        3  2011-01-03       1   0     1        0        1           1   
3        4  2011-01-04       1   0     1        0        2           1   
4        5  2011-01-05       1   0     1        0        3           1   
5        6  2011-01-06       1   0     1        0        4           1   
6        7  2011-01-07       1   0     1        0        5           1   
7        8  2011-01-08       1   0     1        0        6           0   
8        9  2011-01-09       1   0     1        0        0           0   
9       10  2011-01-10       1   0     1        0        1           1   

   weathersit      temp     atemp       hum  windspeed  casual  registered  \
0           2  0.344167  0.363625  0.805833   0.160446     331         654   
1           2  0.363478  0.353739  0.696087   0.248539     131         670   
2           1  0.196364  0.189405  0.437273   0.248309     120        1229   
3           1  0.200000  0.212122  0.590435   0.160296     108        1454   
4           1  0.226957  0.229270  0.436957   0.186900      82        1518   
5           1  0.204348  0.233209  0.518261   0.089565      88        1518   
6           2  0.196522  0.208839  0.498696   0.168726     148        1362   
7           2  0.165000  0.162254  0.535833   0.266804      68         891   
8           1  0.138333  0.116175  0.434167   0.361950      54         768   
9           1  0.150833  0.150888  0.482917   0.223267      41        1280   

    cnt  
0   985  
1   801  
2  1349  
3  1562  
4  1600  
5  1606  
6  1510  
7   959  
8   822  
9  1321

In [69]:
bike_hour_df = pd.read_csv(file_path_hour)
display(bike_hour_df.head())

instant      dteday  season  yr  mnth  hr  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1   0        0        6           0   
1        2  2011-01-01       1   0     1   1        0        6           0   
2        3  2011-01-01       1   0     1   2        0        6           0   
3        4  2011-01-01       1   0     1   3        0        6           0   
4        5  2011-01-01       1   0     1   4        0        6           0   

   weathersit  temp   atemp   hum  windspeed  casual  registered  cnt  
0           1  0.24  0.2879  0.81        0.0       3          13   16  
1           1  0.22  0.2727  0.80        0.0       8          32   40  
2           1  0.22  0.2727  0.80        0.0       5          27   32  
3           1  0.24  0.2879  0.75        0.0       3          10   13  
4           1  0.24  0.2879  0.75        0.0       0           1    1

**Insight:**
- Terdapat fitur dan variabel yang sama pada kedua dataset tersebut (day.csv & hour.csv)

### Assessing Data

### **Menilai data day.csv**

**1. Memahami stuktur data**

In [70]:
bike_day_df.info() # Informasi kolom, tipe data, dan missing values

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     731 non-null    int64  
 1   dteday      731 non-null    object 
 2   season      731 non-null    int64  
 3   yr          731 non-null    int64  
 4   mnth        731 non-null    int64  
 5   holiday     731 non-null    int64  
 6   weekday     731 non-null    int64  
 7   workingday  731 non-null    int64  
 8   weathersit  731 non-null    int64  
 9   temp        731 non-null    float64
 10  atemp       731 non-null    float64
 11  hum         731 non-null    float64
 12  windspeed   731 non-null    float64
 13  casual      731 non-null    int64  
 14  registered  731 non-null    int64  
 15  cnt         731 non-null    int64  
dtypes: float64(4), int64(11), object(1)
memory usage: 91.5+ KB


Terdapat columns yang memiliki tipe data yang kurang tepat, yaitu columns :

* dteday (object yang seharusnya bertipe datetime)
* season, yr, mnth, dan weekday (int64 yang seharusnya bertipe str)

**2. Memeriksa Missing Value**

In [71]:
bike_day_df.isna().sum()

instant       0
dteday        0
season        0
yr            0
mnth          0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64

" Tidak terdapat data yang kosong(Missing Value) pada dataset day.csv"

**3. Memeriksa Duplikat**

In [72]:
bike_day_df.duplicated().sum()

0

" Tidak terdapat duplikat data pada dataset day.csv"

**4. Memeriksa Parameter Statistik**

In [73]:
bike_day_df.describe()

instant      season          yr        mnth     holiday     weekday  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean   366.000000    2.496580    0.500684    6.519836    0.028728    2.997264   
std    211.165812    1.110807    0.500342    3.451913    0.167155    2.004787   
min      1.000000    1.000000    0.000000    1.000000    0.000000    0.000000   
25%    183.500000    2.000000    0.000000    4.000000    0.000000    1.000000   
50%    366.000000    3.000000    1.000000    7.000000    0.000000    3.000000   
75%    548.500000    3.000000    1.000000   10.000000    0.000000    5.000000   
max    731.000000    4.000000    1.000000   12.000000    1.000000    6.000000   

       workingday  weathersit        temp       atemp         hum   windspeed  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean     0.683995    1.395349    0.495385    0.474354    0.627894    0.190486   
std      0.465233    0.544894    0.183051    0.162961    0.142429    0.077498   
min      0.000000    1.000000    0.059130    0.079070    0.000000    0.022392   
25%      0.000000    1.000000    0.337083    0.337842    0.520000    0.134950   
50%      1.000000    1.000000    0.498333    0.486733    0.626667    0.180975   
75%      1.000000    2.000000    0.655417    0.608602    0.730209    0.233214   
max      1.000000    3.000000    0.861667    0.840896    0.972500    0.507463   

            casual   registered          cnt  
count   731.000000   731.000000   731.000000  
mean    848.176471  3656.172367  4504.348837  
std     686.622488  1560.256377  1937.211452  
min       2.000000    20.000000    22.000000  
25%     315.500000  2497.000000  3152.000000  
50%     713.000000  3662.000000  4548.000000  
75%    1096.000000  4776.500000  5956.000000  
max    3410.000000  6946.000000  8714.000000

- Distribusi Normal ,dapat dilihat bahwa mean dan medianya tidak begitu berbeda jauh.

- Tidak terdapat Outlier

- Tidak ada columns yang memiliki jumlah baris yang berbeda

### **Menilai data hour.csv**

**1. Memahami Struktur Data**

In [74]:
bike_hour_df.info() # Informasi kolom dan tipe data

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17379 entries, 0 to 17378
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     17379 non-null  int64  
 1   dteday      17379 non-null  object 
 2   season      17379 non-null  int64  
 3   yr          17379 non-null  int64  
 4   mnth        17379 non-null  int64  
 5   hr          17379 non-null  int64  
 6   holiday     17379 non-null  int64  
 7   weekday     17379 non-null  int64  
 8   workingday  17379 non-null  int64  
 9   weathersit  17379 non-null  int64  
 10  temp        17379 non-null  float64
 11  atemp       17379 non-null  float64
 12  hum         17379 non-null  float64
 13  windspeed   17379 non-null  float64
 14  casual      17379 non-null  int64  
 15  registered  17379 non-null  int64  
 16  cnt         17379 non-null  int64  
dtypes: float64(4), int64(12), object(1)
memory usage: 2.3+ MB


Terdapat columns yang memiliki tipe data yang kurang tepat, yaitu columns :

* dteday (object yang seharusnya bertipe datetime)
* season, yr, mnth, weekday, workingday, dan holiday (int64 yang seharusnya bertipe str)

**2. Memeriksa Missing Value**

In [75]:
bike_hour_df.isna().sum()

instant       0
dteday        0
season        0
yr            0
mnth          0
hr            0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64

" Tidak terdapat data yang kosong(Missing Value) pada dataset day.csv"

**3. Memeriksa Duplikat**

In [76]:
bike_hour_df.duplicated().sum()

0

" Tidak terdapat duplikat data pada dataset day.csv"

**4. Memeriksa Parameter Statistik**

In [77]:
bike_hour_df.describe()

instant        season            yr          mnth            hr  \
count  17379.0000  17379.000000  17379.000000  17379.000000  17379.000000   
mean    8690.0000      2.501640      0.502561      6.537775     11.546752   
std     5017.0295      1.106918      0.500008      3.438776      6.914405   
min        1.0000      1.000000      0.000000      1.000000      0.000000   
25%     4345.5000      2.000000      0.000000      4.000000      6.000000   
50%     8690.0000      3.000000      1.000000      7.000000     12.000000   
75%    13034.5000      3.000000      1.000000     10.000000     18.000000   
max    17379.0000      4.000000      1.000000     12.000000     23.000000   

            holiday       weekday    workingday    weathersit          temp  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.028770      3.003683      0.682721      1.425283      0.496987   
std        0.167165      2.005771      0.465431      0.639357      0.192556   
min        0.000000      0.000000      0.000000      1.000000      0.020000   
25%        0.000000      1.000000      0.000000      1.000000      0.340000   
50%        0.000000      3.000000      1.000000      1.000000      0.500000   
75%        0.000000      5.000000      1.000000      2.000000      0.660000   
max        1.000000      6.000000      1.000000      4.000000      1.000000   

              atemp           hum     windspeed        casual    registered  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.475775      0.627229      0.190098     35.676218    153.786869   
std        0.171850      0.192930      0.122340     49.305030    151.357286   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.333300      0.480000      0.104500      4.000000     34.000000   
50%        0.484800      0.630000      0.194000     17.000000    115.000000   
75%        0.621200      0.780000      0.253700     48.000000    220.000000   
max        1.000000      1.000000      0.850700    367.000000    886.000000   

                cnt  
count  17379.000000  
mean     189.463088  
std      181.387599  
min        1.000000  
25%       40.000000  
50%      142.000000  
75%      281.000000  
max      977.000000

- Distribusi Normal ,dapat dilihat bahwa mean dan medianya tidak begitu berbeda jauh.

- Tidak terdapat Outlier

- Tidak ada columns yang memiliki jumlah baris yang berbeda

**Insight:**
- Kedua dataset tersebut memiliki Penilaian yang sama ,mulai dari Struktur Data,missing value,duplicated, sampai parameter statistik.

### Cleaning Data

 Membersihkan data day.csv dan hour.csv . Berikut tahapan yang akan dilakukan pada cleaning Data :
- Menagani missing values(✅)
- Menagani data duplikat(✅)
- Memeriksa dan menstandarakan format data(❎)

### **Data day.csv dan hour.csv**

Memeriksa dan menstandarkan format data pada :

### 1. dteday (Mengubah tipe data object menjadi tipe data datetime)

In [78]:
# Data day.csv
bike_day_df["dteday"] = pd.to_datetime(bike_day_df["dteday"])
display(bike_day_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   instant     731 non-null    int64         
 1   dteday      731 non-null    datetime64[ns]
 2   season      731 non-null    int64         
 3   yr          731 non-null    int64         
 4   mnth        731 non-null    int64         
 5   holiday     731 non-null    int64         
 6   weekday     731 non-null    int64         
 7   workingday  731 non-null    int64         
 8   weathersit  731 non-null    int64         
 9   temp        731 non-null    float64       
 10  atemp       731 non-null    float64       
 11  hum         731 non-null    float64       
 12  windspeed   731 non-null    float64       
 13  casual      731 non-null    int64         
 14  registered  731 non-null    int64         
 15  cnt         731 non-null    int64         
dtypes: datetime64[ns](1), floa

None

In [79]:
# Data hour.csv
bike_hour_df["dteday"] = pd.to_datetime(bike_hour_df["dteday"])
display(bike_hour_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17379 entries, 0 to 17378
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   instant     17379 non-null  int64         
 1   dteday      17379 non-null  datetime64[ns]
 2   season      17379 non-null  int64         
 3   yr          17379 non-null  int64         
 4   mnth        17379 non-null  int64         
 5   hr          17379 non-null  int64         
 6   holiday     17379 non-null  int64         
 7   weekday     17379 non-null  int64         
 8   workingday  17379 non-null  int64         
 9   weathersit  17379 non-null  int64         
 10  temp        17379 non-null  float64       
 11  atemp       17379 non-null  float64       
 12  hum         17379 non-null  float64       
 13  windspeed   17379 non-null  float64       
 14  casual      17379 non-null  int64         
 15  registered  17379 non-null  int64         
 16  cnt         17379 non-

None

Terlihat bahwa tipe data dteday pada dataset day.csv dan hour.csv sudah berubah menjadi bertipe data datetime

### 2. Season, yr, mnth, dan weekday (Mengubah tipe data int menjadi nilai string yang seharusnya)

1. Mengubah tipe data season yang memiliki value (1, 2, 3, 4), dimana {1:'Spring', 2:'Summer', 3:'Fall', 4:'Winter'}.
2. Mengubah tipe data yr menjadi keterangan tahun.
3. Mengubah tipe data mnth menjadi keterangan bulan.
4. Mengubah tipe data weekday menjadi keterangan hari.

In [80]:
# Mengubah tipe data season
def replacing_season(data_df : pd.core.frame.DataFrame )->pd.core.frame.DataFrame :
  """
   Mengubah nilai numerik pada kolom 'season' menjadi nama musim di DataFrame 'data_df'.

    Fungsi ini memetakan nilai numerik dalam kolom 'season' ke nama musim (Spring, Summer, Fall, Winter)
    menggunakan dictionary mapping. Nilai dalam kolom 'season' akan dimodifikasi secara langsung, dan
    DataFrame yang telah diperbarui dikembalikan sebagai output.

    Mapping yang digunakan:
        1 -> Spring
        2 -> Summer
        3 -> Fall
        4 -> Winter

    Returns:
        pd.core.frame.DataFrame: DataFrame 'data_df' yang telah diperbarui dengan kolom 'season'
        berisi nama musim sebagai string.
  """
  dictionary = {
      1 : "Spring",
      2 : "Summer",
      3 : "Fall",
      4 : "Winter"
  }
  data_df["season"] = data_df["season"].replace(dictionary)
  return data_df

# replacing_season()
# display(bike_day_df)
# display(bike_day_df["season"].value_counts())



In [81]:
# Mengubah tipe data season pada dataset day.csv
bike_day_df = replacing_season(bike_day_df)
display(bike_day_df)
display(bike_day_df["season"].value_counts())

instant     dteday  season  yr  mnth  holiday  weekday  workingday  \
0          1 2011-01-01  Spring   0     1        0        6           0   
1          2 2011-01-02  Spring   0     1        0        0           0   
2          3 2011-01-03  Spring   0     1        0        1           1   
3          4 2011-01-04  Spring   0     1        0        2           1   
4          5 2011-01-05  Spring   0     1        0        3           1   
..       ...        ...     ...  ..   ...      ...      ...         ...   
726      727 2012-12-27  Spring   1    12        0        4           1   
727      728 2012-12-28  Spring   1    12        0        5           1   
728      729 2012-12-29  Spring   1    12        0        6           0   
729      730 2012-12-30  Spring   1    12        0        0           0   
730      731 2012-12-31  Spring   1    12        0        1           1   

     weathersit      temp     atemp       hum  windspeed  casual  registered  \
0             2  0.344167  0.363625  0.805833   0.160446     331         654   
1             2  0.363478  0.353739  0.696087   0.248539     131         670   
2             1  0.196364  0.189405  0.437273   0.248309     120        1229   
3             1  0.200000  0.212122  0.590435   0.160296     108        1454   
4             1  0.226957  0.229270  0.436957   0.186900      82        1518   
..          ...       ...       ...       ...        ...     ...         ...   
726           2  0.254167  0.226642  0.652917   0.350133     247        1867   
727           2  0.253333  0.255046  0.590000   0.155471     644        2451   
728           2  0.253333  0.242400  0.752917   0.124383     159        1182   
729           1  0.255833  0.231700  0.483333   0.350754     364        1432   
730           2  0.215833  0.223487  0.577500   0.154846     439        2290   

      cnt  
0     985  
1     801  
2    1349  
3    1562  
4    1600  
..    ...  
726  2114  
727  3095  
728  1341  
729  1796  
730  2729  

[731 rows x 16 columns]

season
Fall      188
Summer    184
Spring    181
Winter    178
Name: count, dtype: int64

In [82]:
# Mengubah tipe data season pada dataset hour.csv
bike_hour_df = replacing_season(bike_hour_df)
display(bike_hour_df)
display(bike_hour_df["season"].value_counts())

instant     dteday  season  yr  mnth  hr  holiday  weekday  workingday  \
0            1 2011-01-01  Spring   0     1   0        0        6           0   
1            2 2011-01-01  Spring   0     1   1        0        6           0   
2            3 2011-01-01  Spring   0     1   2        0        6           0   
3            4 2011-01-01  Spring   0     1   3        0        6           0   
4            5 2011-01-01  Spring   0     1   4        0        6           0   
...        ...        ...     ...  ..   ...  ..      ...      ...         ...   
17374    17375 2012-12-31  Spring   1    12  19        0        1           1   
17375    17376 2012-12-31  Spring   1    12  20        0        1           1   
17376    17377 2012-12-31  Spring   1    12  21        0        1           1   
17377    17378 2012-12-31  Spring   1    12  22        0        1           1   
17378    17379 2012-12-31  Spring   1    12  23        0        1           1   

       weathersit  temp   atemp   hum  windspeed  casual  registered  cnt  
0               1  0.24  0.2879  0.81     0.0000       3          13   16  
1               1  0.22  0.2727  0.80     0.0000       8          32   40  
2               1  0.22  0.2727  0.80     0.0000       5          27   32  
3               1  0.24  0.2879  0.75     0.0000       3          10   13  
4               1  0.24  0.2879  0.75     0.0000       0           1    1  
...           ...   ...     ...   ...        ...     ...         ...  ...  
17374           2  0.26  0.2576  0.60     0.1642      11         108  119  
17375           2  0.26  0.2576  0.60     0.1642       8          81   89  
17376           1  0.26  0.2576  0.60     0.1642       7          83   90  
17377           1  0.26  0.2727  0.56     0.1343      13          48   61  
17378           1  0.26  0.2727  0.65     0.1343      12          37   49  

[17379 rows x 17 columns]

season
Fall      4496
Summer    4409
Spring    4242
Winter    4232
Name: count, dtype: int64

**keterangan : dapat dilihat bahwa columns season pada kedua dataset sudah diperbaiki menjadi nilai yang sebenarnya**

In [83]:
# Mengubah tipe data yr, mnth, and weekday
def replacing_other(df: pd.DataFrame) -> pd.DataFrame:
    """
    Mengubah kolom 'yr', 'mnth', dan 'weekday' dalam DataFrame berdasarkan kolom 'dteday'.

    Fungsi ini memanfaatkan kolom 'dteday' yang bertipe datetime untuk membuat kolom baru:
        - 'yr': Tahun dari kolom 'dteday'.
        - 'mnth': Nama bulan dari kolom 'dteday'.
        - 'weekday': Nama hari dari kolom 'dteday'.
    Kolom-kolom baru ini ditambahkan atau diperbarui dalam DataFrame yang diberikan.

    Parameter:
        df (pd.DataFrame): DataFrame input yang harus memiliki kolom 'dteday' bertipe datetime.

    Returns:
        pd.DataFrame: DataFrame yang telah diperbarui dengan kolom 'yr', 'mnth', dan 'weekday'.
    """
    # Contoh mengubah kolom 'yr', 'mnth', dan 'weekday' menjadi kategori
    df["yr"] = df["dteday"].dt.year
    df["mnth"] = df["dteday"].dt.month_name()
    df["weekday"] = df["dteday"].dt.day_name()
    return df

In [84]:
# Mengubah tipe data pada dataset day.csv
# Panggil fungsi dan perbarui DataFrame
bike_day_df = replacing_other(bike_day_df)

# Tampilkan DataFrame
display(bike_day_df)

# Tampilkan frekuensi kombinasi nilai pada kolom 'yr', 'mnth', dan 'weekday'
display(bike_day_df[["yr", "mnth", "weekday"]].value_counts())

instant     dteday  season    yr      mnth  holiday    weekday  \
0          1 2011-01-01  Spring  2011   January        0   Saturday   
1          2 2011-01-02  Spring  2011   January        0     Sunday   
2          3 2011-01-03  Spring  2011   January        0     Monday   
3          4 2011-01-04  Spring  2011   January        0    Tuesday   
4          5 2011-01-05  Spring  2011   January        0  Wednesday   
..       ...        ...     ...   ...       ...      ...        ...   
726      727 2012-12-27  Spring  2012  December        0   Thursday   
727      728 2012-12-28  Spring  2012  December        0     Friday   
728      729 2012-12-29  Spring  2012  December        0   Saturday   
729      730 2012-12-30  Spring  2012  December        0     Sunday   
730      731 2012-12-31  Spring  2012  December        0     Monday   

     workingday  weathersit      temp     atemp       hum  windspeed  casual  \
0             0           2  0.344167  0.363625  0.805833   0.160446     331   
1             0           2  0.363478  0.353739  0.696087   0.248539     131   
2             1           1  0.196364  0.189405  0.437273   0.248309     120   
3             1           1  0.200000  0.212122  0.590435   0.160296     108   
4             1           1  0.226957  0.229270  0.436957   0.186900      82   
..          ...         ...       ...       ...       ...        ...     ...   
726           1           2  0.254167  0.226642  0.652917   0.350133     247   
727           1           2  0.253333  0.255046  0.590000   0.155471     644   
728           0           2  0.253333  0.242400  0.752917   0.124383     159   
729           0           1  0.255833  0.231700  0.483333   0.350754     364   
730           1           2  0.215833  0.223487  0.577500   0.154846     439   

     registered   cnt  
0           654   985  
1           670   801  
2          1229  1349  
3          1454  1562  
4          1518  1600  
..          ...   ...  
726        1867  2114  
727        2451  3095  
728        1182  1341  
729        1432  1796  
730        2290  2729  

[731 rows x 16 columns]

yr    mnth       weekday  
2011  April      Friday       5
      November   Wednesday    5
2012  July       Sunday       5
                 Tuesday      5
      June       Friday       5
                             ..
2011  May        Friday       4
      March      Sunday       4
                 Saturday     4
                 Monday       4
2012  September  Wednesday    4
Name: count, Length: 168, dtype: int64

In [85]:
# Mengubah tipe data pada dataset hour.csv
# Panggil fungsi dan perbarui DataFrame
bike_hour_df = replacing_other(bike_hour_df)

# Tampilkan DataFrame
display(bike_hour_df)

# Tampilkan frekuensi kombinasi nilai pada kolom 'yr', 'mnth', dan 'weekhour'
display(bike_hour_df[["yr", "mnth", "weekday"]].value_counts())

instant     dteday  season    yr      mnth  hr  holiday   weekday  \
0            1 2011-01-01  Spring  2011   January   0        0  Saturday   
1            2 2011-01-01  Spring  2011   January   1        0  Saturday   
2            3 2011-01-01  Spring  2011   January   2        0  Saturday   
3            4 2011-01-01  Spring  2011   January   3        0  Saturday   
4            5 2011-01-01  Spring  2011   January   4        0  Saturday   
...        ...        ...     ...   ...       ...  ..      ...       ...   
17374    17375 2012-12-31  Spring  2012  December  19        0    Monday   
17375    17376 2012-12-31  Spring  2012  December  20        0    Monday   
17376    17377 2012-12-31  Spring  2012  December  21        0    Monday   
17377    17378 2012-12-31  Spring  2012  December  22        0    Monday   
17378    17379 2012-12-31  Spring  2012  December  23        0    Monday   

       workingday  weathersit  temp   atemp   hum  windspeed  casual  \
0               0           1  0.24  0.2879  0.81     0.0000       3   
1               0           1  0.22  0.2727  0.80     0.0000       8   
2               0           1  0.22  0.2727  0.80     0.0000       5   
3               0           1  0.24  0.2879  0.75     0.0000       3   
4               0           1  0.24  0.2879  0.75     0.0000       0   
...           ...         ...   ...     ...   ...        ...     ...   
17374           1           2  0.26  0.2576  0.60     0.1642      11   
17375           1           2  0.26  0.2576  0.60     0.1642       8   
17376           1           1  0.26  0.2576  0.60     0.1642       7   
17377           1           1  0.26  0.2727  0.56     0.1343      13   
17378           1           1  0.26  0.2727  0.65     0.1343      12   

       registered  cnt  
0              13   16  
1              32   40  
2              27   32  
3              10   13  
4               1    1  
...           ...  ...  
17374         108  119  
17375          81   89  
17376          83   90  
17377          48   61  
17378          37   49  

[17379 rows x 17 columns]

yr    mnth      weekday  
2011  April     Friday       120
2012  July      Tuesday      120
      May       Thursday     120
2011  July      Friday       120
                Saturday     120
                            ... 
      August    Sunday        89
      February  Tuesday       88
      January   Wednesday     84
                Tuesday       80
                Thursday      79
Name: count, Length: 168, dtype: int64

**keterangan : pada bagian columns yr, mnth, dan weekday sudah diubah nilainya sesuai yang seharusnya . Berlaku pada dataset day.csv dan hour.csv**

In [86]:
bike_day_df.to_csv("first_project_DataAnalyst/dashboard/cleaned_day_df.csv", index=False)
bike_hour_df.to_csv("first_project_DataAnalyst/dashboard/cleaned_hour_df.csv", index=False)

**Insight:**


## Exploratory Data Analysis (EDA)

### Explore **day.csv**

In [87]:
# Melihat rangkuman parameter statistik
bike_day_df.describe(include="all")

instant               dteday season           yr     mnth  \
count   731.000000                  731    731   731.000000      731   
unique         NaN                  NaN      4          NaN       12   
top            NaN                  NaN   Fall          NaN  January   
freq           NaN                  NaN    188          NaN       62   
mean    366.000000  2012-01-01 00:00:00    NaN  2011.500684      NaN   
min       1.000000  2011-01-01 00:00:00    NaN  2011.000000      NaN   
25%     183.500000  2011-07-02 12:00:00    NaN  2011.000000      NaN   
50%     366.000000  2012-01-01 00:00:00    NaN  2012.000000      NaN   
75%     548.500000  2012-07-01 12:00:00    NaN  2012.000000      NaN   
max     731.000000  2012-12-31 00:00:00    NaN  2012.000000      NaN   
std     211.165812                  NaN    NaN     0.500342      NaN   

           holiday   weekday  workingday  weathersit        temp       atemp  \
count   731.000000       731  731.000000  731.000000  731.000000  731.000000   
unique         NaN         7         NaN         NaN         NaN         NaN   
top            NaN  Saturday         NaN         NaN         NaN         NaN   
freq           NaN       105         NaN         NaN         NaN         NaN   
mean      0.028728       NaN    0.683995    1.395349    0.495385    0.474354   
min       0.000000       NaN    0.000000    1.000000    0.059130    0.079070   
25%       0.000000       NaN    0.000000    1.000000    0.337083    0.337842   
50%       0.000000       NaN    1.000000    1.000000    0.498333    0.486733   
75%       0.000000       NaN    1.000000    2.000000    0.655417    0.608602   
max       1.000000       NaN    1.000000    3.000000    0.861667    0.840896   
std       0.167155       NaN    0.465233    0.544894    0.183051    0.162961   

               hum   windspeed       casual   registered          cnt  
count   731.000000  731.000000   731.000000   731.000000   731.000000  
unique         NaN         NaN          NaN          NaN          NaN  
top            NaN         NaN          NaN          NaN          NaN  
freq           NaN         NaN          NaN          NaN          NaN  
mean      0.627894    0.190486   848.176471  3656.172367  4504.348837  
min       0.000000    0.022392     2.000000    20.000000    22.000000  
25%       0.520000    0.134950   315.500000  2497.000000  3152.000000  
50%       0.626667    0.180975   713.000000  3662.000000  4548.000000  
75%       0.730209    0.233214  1096.000000  4776.500000  5956.000000  
max       0.972500    0.507463  3410.000000  6946.000000  8714.000000  
std       0.142429    0.077498   686.622488  1560.256377  1937.211452

Berdasarkan rangkuman parameter statistik di atas, diperoleh jumlah pelanggan dalam kurun waktu 1 tahun terdapat 731 pelanggan .

In [91]:
# Mencari jumlah pelanggan terbanyak diantara 4 musim yang ada berdasarkan cnt (count of total rental bikes including both casual and registered)
season_count_day = bike_day_df.groupby(by = ["season"]).agg({
    "cnt" : "sum"
})
season_count_day

cnt
season         
Fall    1061129
Spring   471348
Summer   918589
Winter   841613

Dari informasi di atas, kita mendapatkan bahwa total pelanggan terbanyak terjadi pada musim Fall(musim gugur), yaitu terdapat : 1.061.129 pelanggan ,selanjutnya total pelanggan terbanyak kedua terjadi pada musim Summer(musim panas), yaitu terdapat : 918.589 pelanggan, selanjutnya total pelanggan terbanyak ketiga terjadi pada musim Winter(musim dingin), yaitu terdapat : 841.613 pelanggan, dan terakhir pelanggan paling sedikit terjadi pada musim Spring(musim semi), yaitu hanya terdapat : 471.348 pelanggan. Informasi ini tentunya dapat memberikan gambaran mengenai perbandingan banyaknya total pelanggan pada setiap musimnya.

In [92]:
# Mencari jumlah pelanggan pada hari kerja dan akhir pekan
work_day_df = bike_day_df.groupby(by = ["weekday"]).agg({
    "cnt" : "sum"
})
work_day_df

cnt
weekday          
Friday     487790
Monday     455503
Saturday   477807
Sunday     444027
Thursday   485395
Tuesday    469109
Wednesday  473048

**Insight:**
- xxx
- xxx

## Visualization & Explanatory Analysis

### Pertanyaan 1:

### Pertanyaan 2:

**Insight:**
- xxx
- xxx

## Analisis Lanjutan (Opsional)

## Conclusion

- Conclution pertanyaan 1
- Conclution pertanyaan 2